SpatialGDK/Source/SpatialGDK/Private/gdk/built_op_list.cc

```diff
@@ -0,0 +1,287 @@
+  opList.ops.emplace_back(op);
+  return *this;
+}
+
+Worker_ComponentData OpListBuilder::StoreComponentData(ComponentData&& data) {
+  opList.dataStorage.emplace_back(std::move(data));
+  auto& storedData = opList.dataStorage.back();
+  return Worker_ComponentData{nullptr, storedData.GetComponentId(), storedData.GetUnderlying(),
+                              nullptr};
+}
+
+Worker_ComponentUpdate OpListBuilder::StoreComponentUpdate(ComponentUpdate&& update) {
+  opList.updateStorage.emplace_back(std::move(update));
+  auto& storedData = opList.updateStorage.back();
+  return Worker_ComponentUpdate{nullptr, storedData.GetComponentId(), storedData.GetUnderlying(),
+                                nullptr};
+}
+
+Worker_CommandRequest OpListBuilder::StoreCommandRequest(CommandRequest&& request) {
+  opList.requestStorage.emplace_back(std::move(request));
+  auto& storedRequest = opList.requestStorage.back();
+  return Worker_CommandRequest{nullptr, storedRequest.GetComponentId(),
+                               storedRequest.GetCommandIndex(), storedRequest.GetUnderlying(),
+                               nullptr};
+}
+
+Worker_CommandResponse OpListBuilder::StoreCommandResponse(CommandResponse&& response) {
+  opList.responseStorage.emplace_back(std::move(response));
+  auto& storedRequest = opList.responseStorage.back();
+  return Worker_CommandResponse{nullptr, storedRequest.GetComponentId(),
+                                storedRequest.GetCommandIndex(), storedRequest.GetUnderlying(),
+                                nullptr};
+}
+
+const std::vector<Worker_Entity>&
+OpListBuilder::StoreQueryEntities(std::vector<EntitySnapshot> entities) {
+  opList.queryEntityStorage.emplace_back(std::vector<Worker_Entity>(entities.size()));
+  for (auto& entity : entities) {
+    Worker_Entity workerEntity;
+    workerEntity.entity_id = entity.GetEntityId();
+
+    // Store the components.
+    auto&& entityState = std::move(entity).ReleaseEntityState();
+    const auto& storedSnapshot = StoreSnapshotComponents(std::move(entityState));
+
+    // Set the entity values from the stored components.
+    workerEntity.component_count = static_cast<std::uint32_t>(storedSnapshot.size());
+    workerEntity.components = storedSnapshot.data();
+
+    // Store the entity.
+    opList.queryEntityStorage.back().emplace_back(workerEntity);
+  }
+  return opList.queryEntityStorage.back();
+}
+
+const std::vector<Worker_ComponentData>&
+OpListBuilder::StoreSnapshotComponents(EntityState entity) {
+  auto& components = entity.GetComponents();
+  opList.snapshotStorage.emplace_back(std::vector<Worker_ComponentData>(components.size()));
+  auto& storedSnapshot = opList.snapshotStorage.back();
+  for (auto& component : components) {
+    storedSnapshot.emplace_back(StoreComponentData(std::move(component)));
+  }
+  return opList.snapshotStorage.back();
+}
+
+const char* OpListBuilder::StoreString(const std::string& string) {
+  opList.stringStorage.emplace_back(string);
```

Comment:
Yeah that sounds like a good idea. It might be worth adding a comment in here that this is likely. There are quite a few places I'm expecting a profiler will say to keep a cache but I decided not to put time into that yet.